src/components/panel/panel.js

```diff
@@ -1101,69 +1129,263 @@ MdPanelPosition.prototype.top = function(opt_top) {
+  if (!this._relativeToRect) {
+    throw new Error('withPanelXPosition can only be used with relative' +
+        'positioning. Set relativeTo first.');
+  }
+
+  var validPosition = false;
+  for (var key in MdPanelPosition.xPosition) {
+    if (MdPanelPosition.xPosition[key] === xPosition) {
+      this._xPosition = xPosition;
+      validPosition = true;
+      break;
+    }
+  }
+
+  if (!validPosition) {
+    throw new Error('withPanelXPosition only accepts the following values:\n' +
+        'center | align-left | align-right | align-start | align-end | ' +
+        'offset-left | offset-right | offset-start | offset-end');
+  }
+
+  return this;
 };
 
 
 /**
- * Sets the value of `right` for the panel.
- * @param {string=} opt_right Value of `right`. Defaults to '0'.
+ * Sets the y position for the panel relative to another element.
+ * yPosition must be one of the MdPanelPosition.yPosition values.
+ *
+ * @param {string} yPosition
  * @returns {MdPanelPosition}
  */
-MdPanelPosition.prototype.right = function(opt_right) {
-  this._right = opt_right || '0';
+MdPanelPosition.prototype.withPanelYPosition = function(yPosition) {
+  if (!this._relativeToRect) {
+    throw new Error('withPanelYPosition can only be used with relative ' +
+        'positioning. Set relativeTo first.');
+  }
+
+  var validPosition;
+  for (var key in MdPanelPosition.yPosition) {
+    if (MdPanelPosition.yPosition[key] === yPosition) {
+      this._yPosition = yPosition;
+      validPosition = true;
+      break;
+    }
+  }
+
+  if (!validPosition) {
+    throw new Error('withPanelYPosition only accepts the following values:\n' +
+        'center | align-tops | align-bottoms | above | below');
+  }
+
   return this;
 };
 
 
 /**
+ * Gets the value of `top` for the panel.
+ * @param {!angular.JQLite} panelEl
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getTop = function(panelEl) {
+  this._calculatePanelPosition(panelEl);
+  return this._top;
+};
+
+
+/**
+ * Gets the value of `bottom` for the panel.
+ * @param {!angular.JQLite} panelEl
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getBottom = function(panelEl) {
+  this._calculatePanelPosition(panelEl);
+  return this._bottom;
+};
+
+
+/**
+ * Gets the value of `left` for the panel.
+ * @param {!angular.JQLite} panelEl
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getLeft = function(panelEl) {
+  this._calculatePanelPosition(panelEl);
+  return this._left;
+};
+
+
+/**
  * Gets the value of `right` for the panel.
+ * @param {!angular.JQLite} panelEl
  * @returns {string}
  */
-MdPanelPosition.prototype.getRight = function() {
+MdPanelPosition.prototype.getRight = function(panelEl) {
+  this._calculatePanelPosition(panelEl);
   return this._right;
 };
+
+
+/**
+ * Calculates the panel position for relative based on the created
+ * panel element.
+ * @param {!angular.JQLite} panelEl
+ */
+MdPanelPosition.prototype._calculatePanelPosition = function(panelEl) {
```

Comment:
FYI: this is a good starting point, but the positioning can get pretty complicated. I'd suggest checkout out the function `attachCalendarPane` in the datepicker to see some of the issue I've run into with positioning the panel next to the trigger.

You may also want to consider: instead of setting one position (like centered-beneath), setting a prioritized list of preferred positions. That way, if the desired position would go off-screen, you can do a more intelligent fallback (eventually falling back to the "this doesn't fit anywhere" case).